src/coreclr/src/jit/morph.cpp

```diff
@@ -8810,23 +8816,14 @@ GenTree* Compiler::fgMorphCall(GenTreeCall* call)
                 lvaGrabTemp(false DEBUGARG("Return value temp for multi-reg return (rejected tail call)."));
             lvaTable[tmpNum].lvIsMultiRegRet = true;
 
-            GenTree* assg = nullptr;
-            if (varTypeIsStruct(call->TypeGet()))
-            {
-                CORINFO_CLASS_HANDLE structHandle = call->gtRetClsHnd;
-                assert(structHandle != NO_CLASS_HANDLE);
-                const bool unsafeValueClsCheck = false;
-                lvaSetStruct(tmpNum, structHandle, unsafeValueClsCheck);
-                var_types structType = lvaTable[tmpNum].lvType;
-                GenTree*  dst        = gtNewLclvNode(tmpNum, structType);
-                assg                 = gtNewAssignNode(dst, call);
-            }
-            else
-            {
-                assg = gtNewTempAssign(tmpNum, call);
-            }
-
-            assg = fgMorphTree(assg);
+            CORINFO_CLASS_HANDLE structHandle = call->gtRetClsHnd;
```

Comment:
@erozenfeld Can you please expain what this part does?

Reply:
It was marking `long` on x86 as `lvIsMultiRegRet ` that was not expected by `decomposeLong` (firing an assert).
Right now decomposeLong has its own mechanism to deal with such lclVars.

A long term solution could be to mark long lclVar as `lvIsMultiRegRet` on x86 because and delete the logic from decomposeLong but for now it is not necessary.